ftl/code/controllers/subsystem/ftl_navigation.dm

```diff
@@ -0,0 +1,189 @@
+SUBSYSTEM_DEF(ftl_navigation)
+	name = "FTL Navigation"
+	wait = 10
+	init_order = INIT_ORDER_FTL_NAV
+	flags = SS_NO_FIRE
+
+	var/ftl_state = FTL_IDLE //is the ship in transit, spooling or idle?
+
+	var/datum/sector/current_sector
+	var/datum/sector/to_sector
+
+	var/list/visited_sectors = list()
+
+	var/list/all_sectors = list()
+
+	var/sector_count = 0
+
+	var/list/ship_areas = list()
+
+	var/list/sector_name_fragments = list("le", "xe", "ge", "za", "ce", "bi", "so", "us", "es", "ar", "ma", "in", "di", "re", "a", "er", "at", "en", "be", "ra", "la", "ve", "ti", "ed", "or", "gu", "an", "te", "is", "ri", "on")
+
+	var/list/sector_types = list()
+	var/list/planet_types = list()
+
+	var/list/nav_consoles = list()
+
+	var/jump_knockdown_force = 3
+	var/jump_throw_force = 0 //Badmin var
+
+
+/datum/controller/subsystem/ftl_navigation/Initialize(timeofday)
+	for(var/type in subtypesof(/datum/planet)) //Get all planet types and weights
+		var/datum/planet/P = type
+		planet_types[type] = initial(P.frequency)
+
+	for(var/type in subtypesof(/datum/sector)) //Get all sector types and weights
+		var/datum/sector/S = type
+		if(initial(S.frequency) > 0)
+			sector_types[type] = initial(S.frequency)
+
+
+	current_sector = new /datum/sector/named/nt_home //Only give them a basic sector to start in
+	visited_sectors += current_sector
+	generate_connecting_sectors(current_sector)
+	for(var/sector_typeless in current_sector.connected_sectors)
+		var/datum/sector/S = sector_typeless
+		generate_connecting_sectors(S)
+
+	var/timer = world.timeofday
+	var/count = 0
+	var/msg
+	for(var/z in SSmapping.levels_by_trait(ZTRAIT_STATION)) //Temp fix to make any map work with ftl transit states
+		for(var/turf/T in block(locate(1,1,z), locate(255,255,z)))
+			if(T.type == /turf/open/space || T.type == /turf/open/space/basic) //I want types only, not subtypes too
+				T.ChangeTurf(/turf/open/space/transit/ftl)

```

Comment:
As I said before you should load the ship as a template onto a blank z level of transit space in the first place. Then you just have noop in the template